### ISSUE TITLE
S4210: FP on static Main method of a net461 class library

### ISSUE THREAD

**Issue by duncanp-sonar** [SonarSource/sonar-dotnet#2428]
Reported by @wouterroos as https://github.com/SonarSource/sonarlint-visualstudio/issues/882

### Description

SonarLint reports an 4210 warning on the following code in a net461 class library

### Repro steps

Please provide the steps required to reproduce the problem

```
    public static class Hosting
    {
        public static void Register()
        {
            //some code.
        }

        private static void Main()
        {
            Register();
            Thread.Sleep(Timeout.Infinite);
        }        
    }
```

### Expected behavior

Since warning 4210 concerns Windows Forms applications, I would not expect this warning to show for a net461 class library that doesn't have any reference to Windows Forms.

### Actual behavior

Warning shows (see image)
![image](https://user-images.githubusercontent.com/5137829/57011217-f0f68b80-6c00-11e9-94ae-19a3cac4e0c8.png)

### Related information

* SonarLint for VS Version 4.9.0.4316
* Visual Studio Version (e.g. Enterprise v15.9.11)
* SonarQube server version not applicable
* Languages in your solution C#

**Comment by Corniel** [comment]
See ##2383 for a way to detect the .NET version.
